src/brod_producer.erl

```diff
@@ -221,11 +232,23 @@ handle_info({msg, Pid, CorrId, #produce_response{} = R},
         "Offset: ~B\n"
         "Error code: ~p",
         [Topic, Partition, Offset, ErrorCode]),
-      {stop, {produce_error, {Topic, Partition, Offset, ErrorCode}}, State};
+      Error = {produce_response_error, Topic, Partition, Offset, ErrorCode},
+      {ok, NewState} = schedule_retry(State, Error),
```

Comment:
and if leader is not available, or the it's not the leader we have connection to, we should re init the socket before schedule retry.
or rather, always re-init the socket (refresh metadata) for each retry?